### PR TITLE
FT2023-341: Addition of the files for configform module .

### PR DESCRIPTION
**Pull request by Arijit-Acharya-innoraft**

### CODE REVIEW

File: web/modules/custom/configform/configform.info.yml [hunk 1]
```diff
@@ -0,0 +1,5 @@
+name: 'Config Form'
+description: 'It contains the tasks of ticket FT2023-341, i.e,creating a configuration file.'  
+type: 'module'
```
Comment (vivekpan27) [inline web/modules/custom/configform/configform.info.yml]: Should this be in quotes?
Same for other values.

File: web/modules/custom/configform/src/Form/CustomConfigForm.php [hunk 2]
```diff
@@ -0,0 +1,150 @@
...
+  /**
+   * Settings variable.
+   */
+  const CONFIGNAME = "configform.settings";
+
+  /**
+   * {@inheritdoc}
+   */
+  public function getFormId() {
+    return 'configform.settings';
```
Comment (vivekpan27) [inline web/modules/custom/configform/src/Form/CustomConfigForm.php]: Please check if we use `.` in form id.

File: web/modules/custom/configform/src/Form/CustomConfigForm.php [hunk 3]
```diff
@@ -0,0 +1,150 @@
...
+   *
+   * @param array $form
+   *   An associative array containing the structure of the form.
+   * @param \Drupal\Core\Form\FormStateInterface $form_state
+   *   The current state of the form.
+   *
+   * @return array
+   *   The form structure.
+   */
+  public function buildForm(array $form, FormStateInterface $form_state) {
```
Comment (vivekpan27) [inline web/modules/custom/configform/src/Form/CustomConfigForm.php:37]: Are you sure you need to add such extensive function docblock for an inherited function?
Same for the other function.

File: web/modules/custom/configform/config/install/configform.settings.yml
Comment (vivekpan27) [inline web/modules/custom/configform/config/install/configform.settings.yml:1]: Config schema file is missing.

File: web/modules/custom/configform/config/install/configform.settings.yml [hunk 4]
```diff
@@ -0,0 +1,8 @@
+full_name :
+  - Arijit Acharya
+phone_no :
+  - 9876543210
+email:
+  - arijit.acharya@innoraft.com
+gender:
+  - Male
```
Comment (vivekpan27) [inline web/modules/custom/configform/config/install/configform.settings.yml]: Why are all the fields arrays instead of being string?

**Review by vivekpan27 — CHANGES_REQUESTED** [review]

File: web/modules/custom/configform/config/install/configform.settings.yml
Comment (vivekpan27) [reply web/modules/custom/configform/config/install/configform.settings.yml:1]: Still missing.

File: web/modules/custom/configform/config/install/configform.settings.yml [hunk 5]
```diff
@@ -0,0 +1,5 @@
+full_name: Arijit Acharya
+phone_no: 9876543210
+email: arijit.acharya@innoraft.com
+gender:
+  - Male
```
Comment (vivekpan27) [inline web/modules/custom/configform/config/install/configform.settings.yml]: Why is this an array?

**Comment by vivekpan27** [comment]
PR is ok.
But default configuration is missing.
Please check how to add default configuration to a module.

**Comment by Arijit-Acharya-innoraft** [comment]
@vivekpan27 The file exists.

**Review by vivekpan27 — APPROVED** [review]

**Comment by vivekpan27** [comment]
Nice, clean PR.

**Pull request merged**